core/vdbe/mod.rs

```diff
@@ -2491,6 +2491,52 @@ impl Program {
                                 state.registers[*dest] = exec_replace(source, pattern, replacement);
                             }
                         },
+                        crate::function::Func::Math(math_func) => match math_func.arity() {
+                            MathFuncArity::Nullary => match math_func {
+                                MathFunc::Pi => {
+                                    state.registers[*dest] =
+                                        OwnedValue::Float(std::f64::consts::PI);
+                                }
+                                _ => {
+                                    unreachable!(
+                                        "Unexpected mathematical Nullary function {:?}",
+                                        math_func
+                                    );
+                                }
+                            },
+
+                            MathFuncArity::Unary => {
+                                let reg_value = &state.registers[*start_reg];
+                                let result = exec_math_unary(reg_value, math_func);
+                                state.registers[*dest] = result;
+                            }
+
+                            MathFuncArity::Binary => {
+                                let lhs = &state.registers[*start_reg];
+                                let rhs = &state.registers[*start_reg + 1];
+                                let result = exec_math_binary(lhs, rhs, math_func);
+                                state.registers[*dest] = result;
+                            }
+
+                            MathFuncArity::UnaryOrBinary => match math_func {
+                                MathFunc::Log => {
+                                    let lhs = &state.registers[*start_reg];
+                                    let rhs = state.registers.get(*start_reg + 1);
+
+                                    let result = if let Some(arg) = rhs {
```

Comment:
sorry for the slow drip of review comments but the presence of the 2nd argument should be taken from the func_ctx arg count, because register `start_reg + 1` may (and likely even will) be populated with something regardless of whether it was allocated for this function

Reply:
Fixed